test-packages/framework/src/framework.ts

```diff
@@ -97,9 +102,89 @@ export class TestFramework implements TestFrameworkAPI {
     }
     const offset = this.getOffset(content);
     this.offset = offset;
-    content = content.replace(/⇶/, "");
+    content = content.replace(new RegExp(CURSOR_ANCHOR, "g"), "");
     await writeFile(filePath, content);
   }
+
+  /**
+   * Updates file content
+   * @param filePath - path of file to update
+   * @param newText - new text to be written to the file
+   * @param options - describes how the file should be updated
+   *    - insertAfter: if provided the text is inserted after that fragment
+   *    - replaceText: if provided the new text replaces that fragment
+   * If options is omitted then new text is added to the end of file
+   */
+  public async updateFileContent(
```

Comment:
The idea is to have a write function which returns void and read function which returns required data. With this, we change that idea. Maybe we should keep that idea to be consistent.

For offset, we have `setter` and `getter` which should holds  up to date data. Now offset is not keep up to date. Pease use ` this.offset = offset;` to assign new offset.

Reply:
I don't like the idea to hold some offset in class property. Offset is returned to consumer right after file update. Why we we need to hold it in framework instance object?

Reply:
Actually I doubt we need a class for the framework as such. There is no internal state in the object. Just projectInfo which is provided by consumer anyway. Rest of the class are just functions, which don't use internal state